pycontrails/core/flight.py

```diff
@@ -938,65 +938,165 @@ def resample_and_fill(
+        fill_method : {"geodesic", "linear"}, optional
+            Choose between ``"geodesic"`` and ``"linear"``, by default ``"geodesic"``.
+            In geodesic mode, large gaps between waypoints are filled with geodesic
+            interpolation and small gaps are filled with linear interpolation. In linear
+            mode, all gaps are filled with linear interpolation.
+        geodesic_threshold : float, optional
+            Threshold for geodesic interpolation, [:math:`m`].
+            If the distance between consecutive waypoints is under this threshold,
+            values are interpolated linearly.
+        nominal_rocd : float | None, optional
+            Nominal rate of climb / descent for aircraft type.
+            Defaults to :attr:`constants.nominal_rocd`.
+        kernel_size : int, optional
+            Passed directly to :func:`scipy.signal.medfilt`, by default 11.
+            Passed also to :func:`scipy.signal.medfilt`
+        cruise_theshold : int, optional
+            Minimal length of time, in seconds, for a flight to be in cruise to apply median filter
+        force_filter: bool, optional
+            If set to true, `filter_altitude` will always be called. otherwise, it will only
+            be called if the flight has a median sample period under 10 seconds
+        drop : bool, optional
+            Drop any columns that are not resampled and filled.
+            Defaults to ``True``, dropping all keys outside of "time", "latitude",
+            "longitude" and "altitude". If set to False, the extra keys will be
+            kept but filled with ``nan`` or ``None`` values, depending on the data type.
+        keep_original_index : bool, optional
+            Keep the original index of the :class:`Flight` in addition to the new
+            resampled index. Defaults to ``False``.
+            .. versionadded:: 0.45.2
+        climb_or_descend_at_end : bool
+            If true, the climb or descent will be placed at the end of each segment
+            rather than the start. Default is false (climb or descent immediately).
 
         Returns
         -------
         Flight
-            Smoothed flight
+            Filled Flight
         """
-        # np.roll pushes the last NaN value from `segment_duration` to the front
-        # so the elapsed time at the first waypoint will be 0
-        seg_dur = self.segment_duration(dtype=np.float64)
-        elapsed_time = np.nancumsum(np.roll(seg_dur, 1))
-        alt_ft = fit_altitude(
-            elapsed_time,
-            self.altitude_ft,
-            max_segments=max_segments,
-            pop=pop,
-            r2_target=r2_target,
-            max_cruise_rocd=max_cruise_rocd,
-            sg_window=sg_window,
-            sg_polyorder=sg_polyorder,
+        clean_flight: Flight
+        # If the flight has a large sampling period, don't try to smooth it unless requested
+        median_gap = np.nanmedian(self.segment_duration())
```

Comment:
You use the segment duration both here and down below when you calculate `max_gap`. I'd suggest storing it as a separate variable above this, then using it here and below.

```python
seg_duration = self.segment_duration()
median_gap = np.nanmedian(seg_duration)
...
max_gap = np.max(seg_duration)
```